script/gen/cxx14

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/env python
+
+""" Extracts MK FFI's specifications from the
+    `include/measurement_kit/README.md` file and generates the corresponding
+    C++11 wrappers into the `include/measurement_kit/cxx14.hpp` file. """
+
+from __future__ import print_function
+
+import json
+import re
+import sys
+
+if sys.version_info[0] == 3:
+    STR = str
+elif sys.version_info[0] == 2:
+    STR = unicode
```

Comment:
I would suggest (since this is dev tooling) to assume just one version of python and not worry about trying to support both.

On which python version did you run this? I would pick that and exit if it doesn't match the desired version.

Reply:
Mostly Python3. I am not sure I would call this a "dev-only" tooling. It is currently run everytime you run `./autogen.sh` to make sure, among other things, that one cannot forget about changes made in the docs and not see the side effects. As such, I wrote it with also Python2 in mind, since there are versions of Ubuntu shipping Python2 as default. (The approach of generating everything on `./autogen.sh` is not something I've married with, but I should not that it did serve me well over the years, because it's one less thing to worry about.)

Reply:
I would suggest not regenerating code when you run autogen.sh if the code is checked into the source tree (which it is).

The generated code may change in subtle and unexpected ways and it can be troublesome to debug and ensure that the user is compiling the same code that is checked into the repo.

Reply:
The problem I want to solve (TM) is: I do not want a developer to apply changes to the format and forgetting to regenerate the `cxx14.hpp` file (or other generated stuff). An alternative solution to solve this problem, rather than generating everything is `autogen.sh`, is to have a script that runs on travis before compiling that (1) regenerates everything and (2) fails if anything that is generated is changed. In principle, this does not necessarily avoid the need to be compatible with Python 2, but seems more consistent as a solution, because it guarantees that, when you compile MK as a user, you get the same code that is checked into the repo.